### PR TITLE
bug 1669248: add dockerhub auth and conform to dockerflow

### PR DESCRIPTION
**Pull request by willkg**
This ditches the deploy-dockerhub.sh script and adds the bits directly
to the CircleCI config file. It also changes it to conform to the
existing Dockerflow conventions.

This switches to using DOCKERHUB_REPO, DOCKER_USER, and DOCKER_PASS
variables. They'll be in the environment when CircleCI is running a job
based on this repo, but won't be available in the environment when
CircleCI is running a job based on a fork.

### CODE REVIEW

**Comment by willkg** [comment]
@sciurus Last one! In CircleCI settings, we need to:

* change DOCKER_USERNAME -> DOCKER_USER
* change DOCKER_PASSWORD -> DOCKER_PASS
* add DOCKERHUB_REPO="mozilla/tecken"

File: .circleci/config.yml [hunk 1]
```diff
@@ -53,15 +75,44 @@ jobs:
 
       - run:
           name: Push to Dockerhub
-          working_directory: ~/tecken
+          working_directory: /antenna
```
Comment (sciurus) [inline .circleci/config.yml:78]:
```suggestion
          working_directory: ~/tecken
```

**Review by sciurus — CHANGES_REQUESTED** [review]

**Comment by sciurus** [comment]
Env var changes are done.

File: .circleci/config.yml (hunk 1 above)
Comment (willkg) [reply .circleci/config.yml:78]: Whoops!

**Review by sciurus — APPROVED** [review]

**Comment by willkg** [comment]
Thank you!

**Pull request merged**